modules/zip/src/lib/tar/tar.ts

```diff
@@ -7,42 +7,39 @@
 
 import {clean, pad, stringToUint8} from './utils';
 import {format} from './header';
+import type {Blocks, Options, Chunks, Chunk} from './types';
 
-let blockSize;
+let blockSize: number;
 const recordSize = 512;
 
 class Tar {
-  /**
-   * @param {number} [recordsPerBlock]
-   */
-  constructor(recordsPerBlock) {
+  written: number;
+  out: Uint8Array;
+  blocks: Blocks;
+  length: number;
+
+  constructor(recordsPerBlock: number) {
     this.written = 0;
     blockSize = (recordsPerBlock || 20) * recordSize;
     this.out = clean(blockSize);
-    /** @type {{ header: any; input: string | Uint8Array; headerLength: number; inputLength: number; }[]} */
     this.blocks = [];
     this.length = 0;
     this.save = this.save.bind(this);
     this.clear = this.clear.bind(this);
     this.append = this.append.bind(this);
   }
 
-  /**
-   * Append a file to the tar archive
-   * @param {string} filepath
-   * @param {string | Uint8Array} input
-   * @param {{ mode?: any; mtime?: any; uid?: any; gid?: any; owner?: any; group?: any; }} [opts]
-   */
+  // Append a file to the tar archive
   // eslint-disable-next-line complexity
-  append(filepath, input, opts) {
-    let checksum;
+  append(filepath: string, input: string | Uint8Array, opts?: Options | undefined) {
+    let checksum: string | any;
 
     if (typeof input === 'string') {
       input = stringToUint8(input);
     } else if (input.constructor && input.constructor !== Uint8Array.prototype.constructor) {
-      const errorInputMatch = input.constructor
-        .toString()
-        .match(/function\s*([$A-Za-z_][0-9A-Za-z_]*)\s*\(/);
+      const errorInputMatch = /function\s*([$A-Za-z_][0-9A-Za-z_]*)\s*\(/.exec(
```

Comment:
I am not sure that we should change this while `ts` refactoring.

Reply:
Yes, less changes are better. There is always a risk of breaking something unexpected and it can take a lot of work for someone else to debug it later. Worst case you can use a `// @ts-ignore` for now.

Reply:
> I am not sure that we should change this while `ts` refactoring.

As far as I know the exec() method tests for a match in a string
so probably eslint does all this dirty work and rewrote the code to make it prettier